app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/room/RoomCategoryTest.kt

```diff
@@ -0,0 +1,214 @@
+package com.github.HumanLearning2021.HumanLearningApp.room
+
+import android.content.Context
+import android.net.Uri
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.hamcrest.CoreMatchers
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.not
+import org.hamcrest.MatcherAssert
+import org.hamcrest.Matchers
+import org.hamcrest.Matchers.contains
+import org.hamcrest.Matchers.hasSize
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.IOException
+import java.util.*
+
+@RunWith(AndroidJUnit4::class)
+class RoomCategoryTest {
+    private lateinit var db: RoomOfflineDatabase
+    private lateinit var categoryDao: CategoryDao
+
+    @Before
+    fun createDb() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        db = Room.inMemoryDatabaseBuilder(context, RoomOfflineDatabase::class.java).build()
+        categoryDao = db.categoryDao()
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun closeDb() {
+        db.close()
+    }
+
+    private fun getRandomString() = "${UUID.randomUUID()}"
+    private fun getRandomCategory(name: String = getRandomString()) = RoomCategory(getRandomString(), name)
+    private fun getRandomPicture(categoryId: String = getRandomString()) = RoomPicture(Uri.parse(getRandomString()), categoryId)
+    private fun getRandomRepresentativePicture(categoryId: String = getRandomString()) = RoomUnlinkedRepresentativePicture(Uri.parse(getRandomString()), categoryId)
+
+    @Test
+    fun insertThenLoadCategories() {
+        val numberOfCategories = (2..50).random()
+        val testCategories = mutableListOf<RoomCategory>()
+        for (i in 0 until numberOfCategories) {
+            testCategories.add(getRandomCategory())
+        }
+
+        categoryDao.insertAll(*testCategories.toTypedArray())
+
+        val res = categoryDao.loadAll()
+
+        MatcherAssert.assertThat(res, Matchers.hasSize(numberOfCategories))
+        MatcherAssert.assertThat(res, Matchers.containsInAnyOrder(*testCategories.toTypedArray()))
+    }
+
+    @Test
+    fun insertThenLoadAllPictures() {
+        val numberOfPictures = (2..10).random()
+        val category = getRandomCategory()
+        val testPictures = mutableListOf<RoomPicture>()
+        for (i in 0 until numberOfPictures) {
+            val pic = getRandomPicture(category.categoryId)
+            testPictures.add(pic)
+        }
+
+        categoryDao.insertAll(category)
+        categoryDao.insertAll(*testPictures.toTypedArray())
+
+        val res = categoryDao.loadAllPictures(category.categoryId).pictures
+
+        MatcherAssert.assertThat(res, Matchers.hasSize(numberOfPictures))
```

Comment:
I would like less prefixing here and on similar lines in this fine but that's stylistic